libseccomp/src/lib.rs

```diff
@@ -617,7 +616,7 @@ impl ScmpFilterContext {
     ///
     /// Accepts file to write to (must be open for writing).
     /// Returns an error if writing to the file fails.
-    pub fn export_pfc(&self, fd: File) -> Result<()> {
+    pub fn export_pfc<T: AsRawFd>(&self, fd: &mut T) -> Result<()> {
```

Comment:
`File::write` (`io::Write::write`) takes a `&mut self`, semantically we do the same (write to the underlying fd). I'm not sure if it would be UB to take a `&T` and write to the underlying fd or if it could cause other issues (a second `&T` at the same time is allowed and `File` is `Send + Snyc`  so even an other thread can have `&T`). But if we take a `&mut T` we are on the safe side (there can not be a `&T` while we have a `&mut T`) and writing `export_pfc(&mut file)` makes it clear that `file` is modified by the function.

Reply:
> if we take a &mut T we are on the safe side (there can not be a &T while we have a &mut T) 
Absolutely! Thank you.